src/EditorFeatures/CSharpTest/InlineDeclaration/CSharpInlineDeclarationTests.cs

```diff
@@ -0,0 +1,347 @@
+        public async Task InlineVariable1()
+        {
+            await TestAsync(
+@"class C
+{
+    void M()
+    {
+        [|int|] i;
+        if (int.TryParse(v, out i))
+        {
+        } 
+    }
+}",
+@"class C
+{
+    void M()
+    {
+        if (int.TryParse(v, out int i))
+        {
+        } 
+    }
+}");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsInlineDeclaration)]
+        public async Task TestMissingInCSharp6()
+        {
+            await TestMissingAsync(
+@"class C
+{
+    void M()
+    {
+        [|int|] i;
+        if (int.TryParse(v, out i))
+        {
+        } 
+    }
+}", parseOptions: CSharpParseOptions.Default.WithLanguageVersion(LanguageVersion.CSharp6));
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsInlineDeclaration)]
+        public async Task InlineVariablePreferVar1()
+        {
+            await TestAsync(
+@"class C
+{
+    void M(string v)
+    {
+        [|int|] i;
+        if (int.TryParse(v, out i))
+        {
+        } 
+    }
+}",
+@"class C
+{
+    void M(string v)
+    {
+        if (int.TryParse(v, out var i))
+        {
+        } 
+    }
+}", options: UseImplicitTypeTests.ImplicitTypeEverywhere());
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsInlineDeclaration)]
+        public async Task InlineVariablePreferVarExceptForPredefinedTypes1()
+        {
+            await TestAsync(
+@"class C
+{
+    void M(string v)
+    {
+        [|int|] i;
+        if (int.TryParse(v, out i))
+        {
+        } 
+    }
+}",
+@"class C
+{
+    void M(string v)
+    {
+        if (int.TryParse(v, out int i))
+        {
+        } 
+    }
+}", options: UseImplicitTypeTests.ImplicitTypeButKeepIntrinsics());
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsInlineDeclaration)]
+        public async Task TestAvailableWhenWrittenAfter1()
+        {
+            await TestAsync(
+@"class C
+{
+    void M()
+    {
+        [|int|] i;
+        if (int.TryParse(v, out i))
+        {
+        }
+        i = 0;
+    }
+}",
+@"class C
+{
+    void M()
+    {
+        if (int.TryParse(v, out int i))
+        {
+        }
+        i = 0;
+    }
+}");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsInlineDeclaration)]
+        public async Task TestMissingWhenWrittenBetween1()
+        {
+            await TestMissingAsync(
+@"class C
+{
+    void M()
+    {
+        [|int|] i;
+        i = 0;
+        if (int.TryParse(v, out i))
+        {
+        }
+    }
+}");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsInlineDeclaration)]
+        public async Task TestMissingWithInitializer()
+        {
+            await TestMissingAsync(
+@"class C
+{
+    void M()
+    {
+        [|int|] i = 0;
+        if (int.TryParse(v, out i))
+        {
+        }
+    }
+}");
+        }
```

Comment:
I now inline if the local has a simple literal initializer or 'default' initializer.